lib/assert_eventually.ex

```diff
@@ -110,6 +110,35 @@ defmodule AssertEventually do
     eventually_impl(assert_call, [], args, timeout)
   end
 
+  defp eventually_impl(assert_call, meta, [{:=, opts, [variable, rest]}], timeout) do
+    ignored_variable = neutralize_variable(variable)
+    neutral_assignment_ast = [{:=, opts, [ignored_variable, {:result, [], AssertEventually}]}]
```

Comment:
Is there any reason we're matching with `result` variable instead of actual `rest` expression? I haven't tested this manually, but I think this implementation doesn't work with `assert_raise` because of that.

Reply:
Actually, `assert_raise` makes no sense along with this use case, but the original question still stands even if this is just my curiosity.

Reply:
I just tested this locally with `neutral_assignment_ast = [{:=, opts, [ignored_variable, rest]}]` and it seems to work just fine while simplifying the code.

Reply:
I think you are right, it is unnecessary :) It's a leftover from me trying to make the code run the right side of assertion just once (#1) but it was solved actually differently. Let me try quickly to fix it (and write a test to check that this is called just once as well).

Reply:
done a6411e9